Project fails to build due to isnan issue in MADP

The project fails to build if MADP is downloaded automatically, with this error trace:
`prob.h:531:30: error: 'isnan' was not declared in this scope, and no declarations were found by argument-dependent lookup at the point of instantiation [-fpermissive]
                     if( isnan( _p[i] ) )
`
This is a common issue when the older code is not up to new C++ standards, and the issue was resolved in the MADP with the latest commit (MADPToolbox/MADP@a6c1bb7), however this project downloads an earlier release of the MADP.

A dirty hack is to edit MADP source files after the download, however the best solution would be to autodownload the latest version od MADP code. At the very least, this information could be added to README of this project.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,30 +36,33 @@ endif()
 
 
 ## MADP toolbox
+set(EXT_DIR "${CMAKE_CURRENT_BINARY_DIR}/third-party/ExternalProject")
+message(STATUS "EXT_DIR: ${EXT_DIR}")
+
 if(MADPPATH)
 	set(MADP_INCLUDE_DIRECTORIES ${MADPPATH}/include)
 	set(MADP_LIBRARY_PATH ${MADPPATH}/lib)
 else()
 	include(ExternalProject)
-	set(EXTERNALS_DIR ${CMAKE_CURRENT_BINARY_DIR}/external)
 	ExternalProject_Add(project_madp
-	    URL               https://github.com/MADPToolbox/MADP/archive/v0.4.1.tar.gz
-	    URL_HASH          MD5=b385c9d4c2b478c0b0611774ee7f26bf
-	    PREFIX            ${EXTERNALS_DIR}/madp
+		GIT_REPOSITORY	   https://github.com/MADPToolbox/MADP
+		GIT_TAG			   a6c1bb7ca18236fc7214d05c6c08dd7d515c4e96
+	    PREFIX            ${EXT_DIR}/madp
 	    BUILD_IN_SOURCE	  1
-	    CONFIGURE_COMMAND <SOURCE_DIR>/configure --prefix=<INSTALL_DIR>
-	    BUILD_COMMAND     make -j12
-	    INSTALL_COMMAND   make install
+	    CONFIGURE_COMMAND sh <SOURCE_DIR>/autogen.sh && <SOURCE_DIR>/configure --prefix=<INSTALL_DIR>
+	    BUILD_COMMAND make
+	    INSTALL_COMMAND make install
 	)
 
-	ExternalProject_Get_Property(project_madp INSTALL_DIR)
-	set(MADP_INCLUDE_DIRECTORIES ${INSTALL_DIR}/include)
+	ExternalProject_Get_Property(project_madp INSTALL_DIR BINARY_DIR SOURCE_DIR)
+    set(MADP_INCLUDE_DIRECTORIES ${INSTALL_DIR}/include ${SOURCE_DIR}/src/include)
 	set(MADP_LIBRARY_PATH ${INSTALL_DIR}/lib)
 
 	message(STATUS "MADP_INCLUDE_DIRECTORIES: ${MADP_INCLUDE_DIRECTORIES}")
-
+	message(STATUS "MADP_LIBRARY_PATH: ${MADP_LIBRARY_PATH}")
 	add_library(MADP STATIC IMPORTED)
 	set_property(TARGET MADP PROPERTY IMPORTED_LOCATION ${MADP_LIBRARY_PATH}/libMADP-0.4.1.so)
+
 	add_dependencies(MADP project_madp)
 	
 endif()

--- a/README.md
+++ b/README.md
@@ -11,8 +11,10 @@ Such Dec-POMDPs can model decentralized information gathering tasks, such as mon
 The algorithm represents policies as finite state controllers, or policy graphs.
 The policy graphs are iteratively improved by finding the action labels of nodes, and the configuration of the out-edges corresponding to observations.
 
-## Blog post
+## Blog post and related work
 You can find a high-level summary of this work from [this blog post](https://laurimi.github.io/research/2020/06/13/active-perception.html).
+
+The follow-up to this work is [Multi-agent active perception with prediction rewards](https://github.com/laurimi/multiagent-prediction-reward) that you may be interested to check out as well.
 
 ## Publications
 For more background information on the algorithm and Dec-POMDPs with information-theoretic rewards, you can refer to the publications below.
@@ -59,7 +61,7 @@ Required system libraries:
 - Boost libraries: program_options regex graph
 
 Required external libraries:
-- [MADP toolbox](https://github.com/MADPToolbox/MADP) v.0.4.1 will be installed automatically, or you can provide a path where to find the library if you have it already by `cmake` options `-DMADPPATH=/path/to/madp`
+- [MADP toolbox](https://github.com/MADPToolbox/MADP)  will be installed automatically, or you can provide a path where to find the library if you have it already by `cmake` options `-DMADPPATH=/path/to/madp`. If you need a different version than the auto-installed one,  modify the `GIT_TAG` in `CMakeLists.txt` or use the `DMADPPATH` variable to point to your preferred installation.
 
 
 ## How to compile

